include/openssl/e_os2.h

```diff
@@ -206,8 +206,12 @@ extern "C" {
 #  endif
 # endif
 
-# if defined(OPENSSL_SYS_UEFI) && !defined(ssize_t)
-#  define ossl_ssize_t int
+# if defined(OPENSSL_SYS_UEFI) && !defined(ossl_ssize_t)
+#  if !defined(ssize_t)
+#   define ossl_ssize_t int
+#  else
+#   define ossl_ssize_t ssize_t
+#  endif
 #  define OSSL_SSIZE_MAX INT_MAX
```

Comment:
> I'd say that if UEFI is available in ILP32 and LP64 flavours, then why not simply stick to long?

Though there is formal objection to doing that. It would formally affect current prototype of our public interfaces that use ossl_ssize_t. And formally this can't be done in minor version release, only major. This leaves already suggested snippet as the only *current* option:
```
# if defined(OPENSSL_SYS_UEFI) && !defined(ossl_ssize_t)
#  define ossl_ssize_t int
#  define OSSL_SIZE_MAX INT_MAX
# endif
```